mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/migration/AttachmentV2Migration.java

```diff
@@ -59,12 +61,12 @@ public Result run() {
 
     private Result migrateAttachment(Attachment attachment) {
         try {
-            blobStore.save(attachment.getBytes())
-                .thenApply(blobId -> CassandraAttachmentDAOV2.from(attachment, blobId))
-                .thenCompose(daoAttachement -> attachmentDAOV2.storeAttachment(daoAttachement).toFuture())
-                .thenCompose(any -> attachmentDAOV1.deleteAttachment(attachment.getAttachmentId()))
-                .join();
-            return Result.COMPLETED;
+            return blobStore.save(attachment.getBytes())
+                .map(blobId -> CassandraAttachmentDAOV2.from(attachment, blobId))
+                .flatMap(attachmentDAOV2::storeAttachment)
+                .then(attachmentDAOV1.deleteAttachment(attachment.getAttachmentId()))
+                .thenReturn(Result.COMPLETED)
+                .block();
```

Comment:
Maybe could use methods like `doOnError` and `onErrorResume` from Reactor to handle the exception and get rid of the `try{} catch{}` blocks?

Reply:
of course.
The main issue with these refactoring is to set a limit. Here, I decided not to handle the exception handling to make the review shorter.

Reply:
Fair enough then !